cmd/geth/main.go

```diff
@@ -463,3 +463,33 @@ func unlockAccounts(ctx *cli.Context, stack *node.Node) {
 		unlockAccount(ks, account, i, passwords)
 	}
 }
+
+// displayDevModeBanner shows some warnings and info for the dev-mode
+func displayDevModeBanner() {
+	log.Info("Starting Geth in ephemeral dev mode...")
+	banner := `
+            _   _                        _
+           | | | |                      | |
+  __ _  ___| |_| |__    ______ ______ __| | _____   __
+ / _' |/ _ \ __| '_ \  |______|______/ _' |/ _ \ \ / /
+| (_| |  __/ |_| | | |              | (_| |  __/\ V /
+ \__, |\___|\__|_| |_|               \__,_|\___| \_/
+  __/ |
+ |___/
+        You are running Geth in --dev mode. Please note the following:
+            1.  This mode is only intended for fast, iterative development
+                without assumptions on security or persistence.
+            2.  The database is created in memory unless specified otherwise.
+                Therefore, shutting down your computer or losing power will wipe
+                your entire block data and chain state for your dev environment.
+            3.  A random, pre-allocated developer account will be available and
+                unlocked as eth.coinbase, which can be used for testing. The
+                random dev account is temporary, stored on a ramdisk, and will
+                be lost if your machine is restarted.
+            4.  Mining is enabled by default. However, the client will only seal
+                blocks if transactions are pending in the mempool. The miner's
+                minimum accepted gas price is 1.
+            5.  Networking is disabled; there is no listen-address, the maximum
+                number of peers is set to 0, and discovery is disabled.`
```

Comment:
Just nitpick that could you add one more blank line at the end? So that we can split banner and geth log clearly.